test/parallel.2.test_helper.bats

```diff
@@ -35,56 +37,60 @@ load 'test_helper/common'
   assert_failure
 }
 
-@test "run_until_success_or_timeout returns instantly on success" {
+@test "${TEST_NAME_PREFIX} run_until_success_or_timeout returns instantly on success" {
   SECONDS=0
   run_until_success_or_timeout 2 true
   [[ ${SECONDS} -le 1 ]]
   assert_success
 }
 
-@test "run_until_success_or_timeout waits for timeout on persistent failure" {
+@test "${TEST_NAME_PREFIX} run_until_success_or_timeout waits for timeout on persistent failure" {
   SECONDS=0
   ! run_until_success_or_timeout 2 false
   [[ ${SECONDS} -ge 2 ]]
   assert_failure
 }
 
-@test "repeat_in_container_until_success_or_timeout fails immediately for non-running container" {
+@test "${TEST_NAME_PREFIX} repeat_in_container_until_success_or_timeout fails immediately for non-running container" {
   SECONDS=0
   ! repeat_in_container_until_success_or_timeout 10 name-of-non-existing-container true
   [[ ${SECONDS} -le 1 ]]
 }
 
-@test "repeat_in_container_until_success_or_timeout run command in container" {
+@test "${TEST_NAME_PREFIX} repeat_in_container_until_success_or_timeout run command in container" {
   local CONTAINER_NAME
   CONTAINER_NAME=$(docker run --rm -d alpine sleep 100)
   SECONDS=0
   ! repeat_in_container_until_success_or_timeout 10 "${CONTAINER_NAME}" sh -c "echo '${CONTAINER_NAME}' > /tmp/marker"
   [[ ${SECONDS} -le 1 ]]
   run docker exec "${CONTAINER_NAME}" cat /tmp/marker
   assert_output "${CONTAINER_NAME}"
+
+  docker rm -f "${CONTAINER_NAME}"
 }
 
-@test "container_is_running" {
+@test "${TEST_NAME_PREFIX} container_is_running" {
   local CONTAINER_NAME
   CONTAINER_NAME=$(docker run --rm -d alpine sleep 100)
   container_is_running "${CONTAINER_NAME}"
   docker rm -f "${CONTAINER_NAME}"
   ! container_is_running "${CONTAINER_NAME}"
 }
 
-@test "wait_for_smtp_port_in_container aborts wait after timeout" {
+@test "${TEST_NAME_PREFIX} wait_for_smtp_port_in_container aborts wait after timeout" {
   local CONTAINER_NAME
   CONTAINER_NAME=$(docker run --rm -d alpine sleep 100)
   SECONDS=0
   TEST_TIMEOUT_IN_SECONDS=2 run wait_for_smtp_port_in_container "${CONTAINER_NAME}"
   [[ ${SECONDS} -ge 2 ]]
   assert_failure
   assert_output --partial "Timed out on command"
+
+  docker rm -f "${CONTAINER_NAME}"container_has_service_running/wait_for_service
```

Comment:
What is the need for this line here?

The container should remove itself once done due to the `docker run --rm`?

Reply:
Ah yes, this was due to me testing repeatedly, and I did not want to have that many containers running on my small machine. Can be removed :)